[RHACS] ROX-19656: Documenting updates to build-time network policy tools (#64628)
openshift/openshift-docs: operating/manage-network-policies.adoc

include::modules/enable-alert-on-baseline-violations-ng20.adoc[leveloffset=+2]

//About build time network policy tools
[id="about-build-time-network-policy-tools_{context}"]
== About build time network policy tools

Build time network policy tools let you automate the creation and validation of Kubernetes network policies in your development and operations workflows using the `roxctl` CLI. These tools work with a specified file directory containing your project's workload and network policy manifests and do not require {product-title-short} authentication.

.Network policy tools
[%autowidth]
|===
| *Command* | *Description*

|`roxctl netpol generate`

> |Generates Kubernetes network policies by analyzing your project's YAML manifests in a specified directory, and was previously known as `roxctl generate netpol`. For more information, see xref:../operating/manage-network-policies.adoc#using-the-build-time-network-policy-generator[Using the build-time network policy generator].

Reviewer: ```suggestion
|Generates Kubernetes network policies by analyzing your project's YAML manifests in a specified directory. The previous syntax was `roxctl generate netpol`. For more information, see xref:../operating/manage-network-policies.adoc#using-the-build-time-network-policy-generator[Using the build-time network policy generator].
```

|`roxctl netpol connectivity map`

> |Lists the allowed connections between workloads in your project directory, considering the workload and Kubernetes network policy manifests. Additionally, a graphical representation in `.dot` format is included. For more information, see xref:../operating/manage-network-policies.adoc#connectivity-mapping-using-the-roxctl-connectivity-map-command_manage-network-policies[Connectivity mapping using the roxctl connectivity-map command].

Reviewer: ```suggestion
|Lists the allowed connections between workloads in your project directory by examining the workload and Kubernetes network policy manifests. Additionally, a graphical representation in `.dot` format is included. For more information, see xref:../operating/manage-network-policies.adoc#connectivity-mapping-using-the-roxctl-connectivity-map-command_manage-network-policies[Connectivity mapping using the roxctl connectivity-map command].
```
